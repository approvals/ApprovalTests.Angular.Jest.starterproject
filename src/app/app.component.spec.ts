import { describe, expect } from "@jest/globals";
import { verify } from "approvals/lib/Providers/Jest/JestApprovals";
import { Options } from "approvals/lib/Core/Options";
import { AppComponent } from "./app.component";
import { ComponentFixture, TestBed } from '@angular/core/testing';

describe('ApprovalsStuff', () => {

  it('should approve', () => {
    expect("5").toBe("5");
  });

  it('should approve with approvals', () => {
    verify('foo');
  });
});

describe('AppComponent', () => {
  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ AppComponent ],
      // ... any other necessary modules or providers
    })
    .compileComponents();
    fixture = TestBed.createComponent(AppComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should render the expected HTML', () => {
    const compiled = fixture.debugElement.nativeElement;
    const htmlContent = compiled.outerHTML;
    // Your test assertion. For instance:
    const options = new Options().forFile().withFileExtention(".html")
    verify(htmlContent, options);    
  });
});
