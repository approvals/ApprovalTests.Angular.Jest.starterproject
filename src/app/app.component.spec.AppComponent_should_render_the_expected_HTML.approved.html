<div id="root0" ng-version="16.2.4"><div>works</div></div>
